test/framework/src/main/java/org/elasticsearch/index/mapper/MapperTestCase.java

```diff
@@ -171,6 +171,16 @@ protected static void assertNoDocValuesField(ParseContext.Document doc, String f
         }
     }
 
+    protected void assertDimension(boolean isDimension, Consumer<MappedFieldType> checker) throws IOException {
```

Comment:
Yeah, you're right, this one is pretty ugly. If you passed a `Function<MappedFieldType, Boolean>` I think it'd be a little more readable. But still kind of messy to call. We have other code that hides this with an unchecked cast - I'm not sure that's better. But it'd look something like:
```
protected <T> void assertDimension(boolean isDimension, Function<T, Boolean> isDimension) throws IOException {
     MapperService mapperService = ....;
     @SuppressWarnings("unchecked") // Syntactic sugar in tests
     T fieldType = (T) mapperService.fieldType("field");
     assertThat(isDimension.apply(fieldType), equalTo(isDimensions));
}
```

You cal it like:
```
    assertDimension(true, NumberFieldType::isDimension);
```

I think it'd be *shorter* but maybe less clear.